### ISSUE TITLE
fatal error: charconv: No such file or directory  #include <charconv>

### ISSUE THREAD

**Issue by leoluopy** [onnx/onnx-mlir#2007]
while build onnx-mlir , i met the error below:

fatal error: charconv: No such file or directory #include <charconv>
FrontendDialectTransformer.cpp:161:28: error: no match for ‘operator[]’ (operand types are ‘const google::protobuf::RepeatedPtrField<onnx::TensorShapeProto_Dimension>’ and ‘int’)
       if (shape_proto.dim()[i].dim_value()) {
                            ^
/home/algo_share/projects/peiyuanluo/workspace/onnx-mlir/src/Builder/FrontendDialectTransformer.cpp:162:49: error: no match for ‘operator[]’ (operand types are ‘const google::protobuf::RepeatedPtrField<onnx::TensorShapeProto_Dimension>’ and ‘int’)
         int dim_numeric_size = shape_proto.dim()[i].dim_value();
                                                 ^
/home/algo_share/projects/peiyuanluo/workspace/onnx-mlir/src/Builder/FrontendDialectTransformer.cpp: In member function ‘void onnx_mlir::detail::FrontendGenImpl::ImportNodeGeneric(const onnx::NodeProto&)’:
/home/algo_share/projects/peiyuanluo/workspace/onnx-mlir/src/Builder/FrontendDialectTransformer.cpp:425:49: error: no match for ‘operator[]’ (operand types are ‘const google::protobuf::RepeatedPtrField<std::__cxx11::basic_string<char> >’ and ‘int’)
       frontend_symbols_.AddMapping(node.output()[i], r);

it seems like , it's a repo branch mismatch problem . 
now i building onnx-mlir at ubuntu 18.04 . 
onnx-mlir branch is : main
llvm-project bracnh is : (HEAD detached at ba8b8a73fcb6)  
 ba8b8a73fcb6b830e63cd8e20c6e13b2a14d69bf

third_party directory are already removed ant then reinited by command " git submodule update --init --recursive " 

so any one has any idea ? 

building command is following the page of :
https://github.com/onnx/onnx-mlir/blob/main/docs/BuildOnLinuxOSX.md#mlir

**Comment by hamptonm1** [comment]
Hi @leoluopy and welcome!!!!! Seems like you are on the right commit for llvm-project. When I usually face this issue I rebuild the `build` directory in `onnx-mlir`. Basically nuke the directory and follow the instructions in the documentation to use either `ninja` or `cmake` to recompile.

**Comment by leoluopy** [comment]
@hamptonm1 , thanks for reply . 
   i tried to clean all the onn-mlir/build/  directory , first make sure onnx-mlir/build is an empty directory then run the following command to reconfigure and rebuild the onnx-mlir . 

cmake -G Ninja         -DCMAKE_CXX_COMPILER=/usr/bin/c++         -DMLIR_DIR=${MLIR_DIR}         ..

cmake --build .

but , it didn't work . the build process failed at : [582/647]  .
error is the same:
CompilerUtils.cpp:39:10: fatal error: charconv: No such file or directory
 #include <charconv>
FrontendDialectTransformer.cpp:161:28: error: no match for ‘operator[]’

do you have any other idea ?

**Comment by tungld** [comment]
This `charconv` requires gcc > 8.3. Which gcc version are you using?

**Comment by gongsu832** [comment]
`charconv` is a c++17 feature which gcc/g++ does support since version 5. However, the header file only existed starting from libstdc++ 8. Ubuntu 18.04 (Bionic) comes with gcc/g++ 7 and libstdc++ 7 so that's why you get the error.

You could manually install gcc/g++ 8 and libstdc++ 8 in Bionic so you can compile `CompilerUtils.cpp`. But even then, you will also need to manually install python 3.8 since Bionic comes with 3.6 which is EOL and we don't support.

So in short, Bionic is too old and you should try a more recent version such as Focal (the one we use in our build).

**Comment by leoluopy** [comment]
@tungld @gongsu832
thanks for replying , i'll try it.
